packages/desktop/views/onboarding/views/complete-onboarding/CompleteOnboardingRouterView.svelte

```diff
@@ -7,10 +7,22 @@
     import EnterNameView from './views/EnterNameView.svelte'
     import EnterPinView from './views/EnterPinView.svelte'
     import FinishOnboardingView from './views/FinishOnboardingView.svelte'
+    import { isStrongholdUnlocked } from '@core/profile-manager'
+    import { RestoreProfileType, onboardingProfile, restoreBackupFromStrongholdFile } from '@contexts/onboarding'
 
     $: if (features.analytics.onboardingRoute.enabled && $completeOnboardingRoute) {
         Platform.trackEvent('complete-onboarding-route', { route: $completeOnboardingRoute })
     }
+    $: verifyAndRestoreStronghold(), $completeOnboardingRoute
+
+    async function verifyAndRestoreStronghold() {
+        if ($onboardingProfile?.restoreProfileType === RestoreProfileType.Stronghold) {
+            const _isStrongholdUnlocked = await isStrongholdUnlocked()
+            if (!_isStrongholdUnlocked) {
+                await restoreBackupFromStrongholdFile($onboardingProfile.strongholdPassword)
+            }
+        }
+    }
```

Comment:
`restoreBackupFromStrongholdFile` is quite intense, i think we could just update the password, wydt?
`import { setStrongholdPassword } from '@core/profile-manager'`
```suggestion
    // Note: needed to cover the cases where the user has waited so long that the stronghold is locked
    $: $completeOnboardingRoute, unlockStronghold()
    function unlockStronghold(): void {
        if ($onboardingProfile?.restoreProfileType === RestoreProfileType.Stronghold) {
            void setStrongholdPassword($onboardingProfile.strongholdPassword)
        }
    }
```

Reply:
Also, I was thiking we dont need this logic in all these routes, its only important in the final view, right before doing all the logic
![image](https://github.com/iotaledger/firefly/assets/3624944/34111807-ba5d-4f7a-b324-44fd270f5ba5)